docs/api.md

```diff
@@ -21,39 +21,73 @@ type HostedClusterSpec struct {
     ClusterID ClusterID `json:"clusterID"`
 
     Release Release
-    
+
+    Networking ClusterNetworking
+
     // NOTE: This might not make sense as control plane
     // inputs can be specific to versions
     ControlPlane ControlPlaneSpec
-    
+
     // PullSecret is propagated to the container runtime
     // of any nodes associated with this cluster.
     PullSecret LocalObjectReference
+
+    // ProviderCreds is a reference to a secret containing cloud account info
+    ProviderCreds LocalObjectReference
+
+    // Platform contains platform-specific details for
+    // the cluster
+    Platform PlatformSpec `json:"platform"`
+
+    // InfraID is used to identify the cluster in cloud platforms
+    InfraID string `json:"infraID,omitempty"`
```

Comment:
can you remind me why we need this exposed in the spec?

Reply:
@enxebre originally I had this in the status because it would be auto-generated by the hosted cluster controller, then I remembered that at some point we will likely need a way to restore a HostedCluster cr for a cluster that already exists (for DR). If not included in the spec, we won't be able to do this.

Reply:
Make sense, though may be wait to bake into the API until we have the story for DR fleshed out to avoid confusion